src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs

```diff
@@ -172,22 +172,14 @@ private PSVirtualNetworkGatewayConnection CreateVirtualNetworkGatewayConnection(
             vnetGatewayConnection.ConnectionType = this.ConnectionType;
             vnetGatewayConnection.RoutingWeight = this.RoutingWeight;
             vnetGatewayConnection.SharedKey = this.SharedKey;
-            
-            if (!string.IsNullOrEmpty(this.EnableBgp))
-            {
-                vnetGatewayConnection.EnableBgp = bool.Parse(this.EnableBgp);
-            }
-            else
-            {
-                vnetGatewayConnection.EnableBgp = false;
-            }
+
+            vnetGatewayConnection.EnableBgp = this.EnableBgp;
```

Comment:
Default values for the Booleans are false anyways, so directly setting in New-cmd should be good. I am interested in using the MyInvocation for the Set-cmd since it can allow us to keep the same parameter type across both new and set.